### ISSUE TITLE
React 16 support?

### ISSUE THREAD

**Issue by kesha-antonov** [chenglou/react-motion#456]
Hello!

RN 0.43 require to use react-16.

It'll be great to update this lib to use React 16.
Can we do that?

**Comment by nkbt** [comment]
Should be compatible now. The only annoyance is not matching peer
dependency but that is not a blocker.

It makes sense to add react 16 to peer deps tho. Together with es6 classes
and publish major release

-- 
Nik Butenko
nik@butenko.me
http://github.com/nkbt

**Comment by nkbt** [comment]
I checked and it looks like we still can maintain older versions of React as peerDeps, so it should be safe to add v16 to the peers without major release.

**Comment by kesha-antonov** [comment]
@nkbt Nice! Thanks!

**Comment by MichaelDeBoey** [comment]
`React v16.0` isn't released yet, but everything that works in `v15.5.0`, [will keep working](https://facebook.github.io/react/blog/#new-deprecation-warnings) in `v16.0`, so this can be closed I think :-)

CC: @nkbt

**Comment by sammkj** [comment]
@MichaelDeBoey what about `<TransitionMotion />`? Do we still need a `<div />` to wrap child components?

**Comment by bezreyhan** [comment]
Do we know when the support for React 16 will be released. I'm getting the following error after upgrading to React 16:

`Cannot read property 'objectOf' of undefined`

The error is happening in `Motion.js:44` because of PropTypes:

```
defaultStyle: _react.PropTypes.objectOf(_react.PropTypes.number)
```

**Comment by nkbt** [comment]
Motion with `prop-types` was release quite a long time ago.

**Comment by bezreyhan** [comment]
I realized the problem is coming from swagger-ui which is using an older version of react-collapse v2.3.1

Sorry about that.

**Comment by nikglavin** [comment]
Are you able to tag a version from Master that "supports" React v16?

`"react-motion": "0.5.1",`
`error react-motion#react@^0.14.9 || ^15.3.0 doesn't satisfy found match of react@16.0.0`

This has been resolved on the master branch.

**Comment by nkbt** [comment]
I've published 0.5.2 with react 16 peerDependency, give it a shot!
